appveyor/numpy.patch

```diff
@@ -0,0 +1,831 @@
+-                    if os.path.abspath(f)==os.path.abspath(t):
++                    if os.path.abspath(f) == os.path.abspath(t):
+                         continue
+                     if os.path.isfile(t):
+                         os.remove(t)
+                     try:
+                         self.move_file(f, module_build_dir)
+                     except DistutilsFileError:
+-                        log.warn('failed to move %r to %r' \
++                        log.warn('failed to move %r to %r'
+                                  % (f, module_build_dir))
+ 
+             if f_sources:
+@@ -278,11 +319,49 @@ class build_clib(old_build_clib):
+         else:
+             f_objects = []
+ 
+-        objects.extend(f_objects)
+-
+-        # assume that default linker is suitable for
+-        # linking Fortran object files
+-        compiler.create_static_lib(objects, lib_name,
++        libs = []
++
++        if objects:
++            # Go ahead and compile the c code
++            compiler.create_static_lib(objects, lib_name + '_msvc',
++                                       output_dir=self.build_clib,
++                                       debug=self.debug)
++            libs += [os.path.join(self.build_clib, lib_name + '_msvc.lib')]
++
++        if f_objects:
++            if lib_name == 'odepack':
++                f_objects = [
++                    obj for obj in f_objects if 'xsetf' not in obj and 'xsetun' not in obj]
++
++            # link the fortran sources with gfortran
++            fcompiler.link_shared_object(f_objects, lib_name + '_gfortran.dll',
++                                         output_dir=self.build_clib,
++                                         library_dirs=[
++                                             self.build_clib,
++                                             '.',
++                                         ],
++                                         extra_postargs=self.dlls + [
++                                             '-Wl,--out-implib,' + lib_name + '.dll.a',
++                                             '-Wl,--output-def,' + lib_name + '_gfortran.def',
++                                             '-Wl,--export-all-symbols',
++                                             '-Wl,--enable-auto-import',
++                                         ],
++                                         debug=self.debug)
++
++            # Create the gfortran lib
++            specifier = '/MACHINE:X64'
++            lib_args = ['/def:{}_gfortran.def'.format(lib_name),
++                        '/OUT:{}_gfortran.lib'.format(lib_name), specifier]
++            if not compiler.initialized:
++                compiler.initialize()
++            compiler.spawn([compiler.lib] + lib_args)
++
++            libs += [lib_name + '_gfortran.lib']
++            self.dlls.append(os.path.join(
++                self.build_clib, lib_name + '_gfortran.dll'))
++
```

Comment:
Each time we build a DLL, we need to keep track of it so that dependendent fortran code can link to the DLL.